onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc

```diff
@@ -686,6 +694,7 @@ Status QnnBackendManager::LoadCachedQnnContextFromBuffer(char* buffer, uint64_t
 }
 
 Status QnnBackendManager::SetupBackend(const logging::Logger& logger, bool load_from_cached_context) {
+  std::lock_guard<std::mutex> lock(logger_mutex_);
```

Comment:
this one is not required since you checked backend_setup_completed_ in ResetQnnLogLevel()

Reply:
We should move the codes that create the ETW Callback to the place after the QnnBackendManager created. https://github.com/microsoft/onnxruntime/blob/fc2be09386fe8c195c224b1cbb5b15a1277e0209/onnxruntime/core/providers/qnn/qnn_execution_provider.cc#L432-L442